test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java

```diff
@@ -82,6 +82,7 @@
     public static final String CLIENT_RETRY_TIMEOUT = "client.retry.timeout";
     public static final String CLIENT_SOCKET_TIMEOUT = "client.socket.timeout";
     public static final String CLIENT_PATH_PREFIX = "client.path.prefix";
+    public static final String STRICT_DEPRECATION_MODE = "strict.deprecation.mode";
```

Comment:
I'd save this for a follow up change. I suspect we'll want to control this some other way in tests. We might be able to get away with a method on `ESRestTestCase` that returns the default for it but I'm not sure. Either way, I think it should wait.

Reply:
OK, I am removing it. My thinking here was to introduce the hook even if not used ;)
A method could be a way to go, I was following the way other tests are defining their settings (i.e. socket time out). But let's table this till the next PR ;)